### PR TITLE
Add facts and conditional logic for RL6 and RL10 support

### PR DESCRIPTION
**Pull request by nrvale0**
Facts: is_rightscale(Boolean), rightlink_version(String), rightlink_maj_version(Integer).

I did *not* update the README.md re RL10.

There are lots of changes here...let me know if you think I should squash.

@diranged @siminm

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+*~
+.#*
+\#.*
+*.lock
+spec/fixtures/modules
+.travis.yml.swp
```
Comment (siminm) [inline .gitignore]: Just add *.swp or configure your vim to use a dedicated swap directory

File: spec/unit/facter/#is_rightscale_spec.rb# [hunk 2]
```diff
@@ -0,0 +1,48 @@
+require 'spec_helper'
```
Comment (siminm) [inline spec/unit/facter/#is_rightscale_spec.rb#]: What is this file...???

File: .gitignore (hunk 1 above)
Comment (diranged) [inline .gitignore]: I'd change this to `*.swp`

**Comment by siminm** [comment]
Your .rb files are missing file docs. I don't know what those are supposed to do

File: lib/facter/rightlink_version.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  setcode do
+    version = nil
+    out = nil
+
+    # check for RightLink == 10
+    # regex = %r{^\w+ (?<version>\d+\.\d+\.\d+).*$}
+    regex = "^\w+ \d+\.\d+\.\d+.*$"
+    out = Facter::Util::Resolution.exec('rightlink --version')
+    Facter.debug("rightlink_version :: Match status for RightLink 10: %s" % (out.to_s))
+    unless nil != out
```
Comment (diranged) [inline lib/facter/rightlink_version.rb]: Double negative .. isnt this the same as `if nill != out`?

File: lib/facter/rightlink_version.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      # check for RightLink < 10
+      # regex = %r{^rs_config (?<version>\d\.\d+\.\d+).*$}
+      regex = "^rs_config \d+\.\d+\.\d+.*$"
+      out = Facter::Util::Resolution.exec('rs_config --version')
+      Facter.debug("rightlink_version :: Match status for RightLink < 10: %s" % (out.to_s))
+    end
+
+    unless nil == out
```
Comment (diranged) [inline lib/facter/rightlink_version.rb]: isnt this the same as `if nil != out`?

File: manifests/init.pp [hunk 5]
```diff
@@ -21,23 +21,42 @@
...
-    'rest-client':
-    ensure   => $rest_client,
-    provider => 'gem';
+  ) inherits ::rightscale::params {
 
-    'right_aws':
-    ensure   => $right_aws,
-    provider => 'gem',
-    require  => Package['rest-client'];
+  # pretty sure our version of stdlib does not include validate_integer
```
Comment (diranged) [inline manifests/init.pp]: this comment is no longer valid, right?

File: spec/unit/facter/#is_rightscale_spec.rb# (hunk 2 above)
Comment (diranged) [inline spec/unit/facter/#is_rightscale_spec.rb#]: Whats up with the filename -- are the hashes intentional?

Comment (diranged) [inline spec/unit/facter/#is_rightscale_spec.rb#]: Er -- I think this is a bogus file now that I scroll further down?

File: spec/unit/provider/rs_tag/ruby_spec.rb [hunk 6]
```diff
@@ -35,7 +36,7 @@
 
   describe 'exists?' do
     it 'checks if `test` tag exists (should not)' do
-       provider.exists?.should be_false
+       provider.exists?.should be_falsey
```
Comment (diranged) [inline spec/unit/provider/rs_tag/ruby_spec.rb]: false`y` ?

**Comment by diranged** [comment]
Overall just a few quick questions and nits.. this looks basically functional though. The one thing I notice is that there isn't yet support in here for using `rsc` to get the system tags and other facts (see the `rs_facts.rb` file).

We don't need that to merge this, but we will ultimately need it to fully support all of our puppet code.

File: .gitignore (hunk 1 above)
Comment (nrvale0) [inline .gitignore]: fixed

File: spec/unit/facter/#is_rightscale_spec.rb# (hunk 2 above)
Comment (nrvale0) [inline spec/unit/facter/#is_rightscale_spec.rb#]: fixed

File: manifests/init.pp (hunk 5 above)
Comment (nrvale0) [inline manifests/init.pp]: Unfortunately it is still valid. I wasn't about to undertake an upgrade of stdlib as part of this feature branch though I would def recommend it as another feature branch/PR.

File: spec/unit/provider/rs_tag/ruby_spec.rb (hunk 6 above)
Comment (nrvale0) [inline spec/unit/provider/rs_tag/ruby_spec.rb]: falsey == '', nil, boolean false, ....
false == boolean false

and it varies from Puppet major versions. It was failing across different versions of the test matrix until I loosed up the def of false just a little bit.

**Comment by nrvale0** [comment]
@diranged re rsc + RL10...there's considerable addtl work for full RL10 compat.

File: lib/facter/rightlink_version.rb (hunk 3 above)
Comment (nrvale0) [inline lib/facter/rightlink_version.rb]: fixed

**Pull request merged**